packages/blocks/src/BlocksComponent.tsx

```diff
@@ -4,42 +4,53 @@ import type { BlocksProps, Cells } from './types';
 
 export const BlocksComponent = forwardRef(
   <TCell extends Cells>(
-    { cells, blocks, children, Block = Fragment }: BlocksProps<TCell>,
+    { cells, blocks, children, Block }: BlocksProps<TCell>,
     ref: ForwardedRef<typeof Block>,
   ) => {
+    if (!Array.isArray(blocks) || !blocks?.length) {
+      return null;
+    }
+
     return (
       <>
-        {Array.isArray(blocks) &&
-          !!blocks?.length &&
-          blocks?.map((block, index) => {
+        {blocks?.map((block, index) => {
+          if (!Array.isArray(block) || !block?.length) {
+            return null;
+          }
+
+          const cellsList = block?.map(({ type, ...cell }) => {
+            const Cell = cells[type];
+
             return (
-              <Block
-                key={`block:${index}`}
-                block={block}
-                // @ts-expect-error - `Fragment` does not have a `ref` prop.
-                ref={ref}
+              <Fragment
+                key={`${(cell[cell?.keyProp as keyof typeof cell] as string) ?? ''}:${
+                  (cell?.id as string) ?? ''
+                }`}
               >
-                {Array.isArray(block) &&
-                  !!block?.length &&
-                  block?.map(({ type, ...cell }) => {
-                    const Cell = cells[type];
-
-                    return (
-                      <Fragment
-                        key={`${(cell[cell?.keyProp as keyof typeof cell] as string) ?? ''}:${
-                          (cell?.id as string) ?? ''
-                        }`}
-                      >
-                        {
-                          // @ts-expect-error - expects the consumer of `@ballerine/blocks` to register `Blocks.cells` for type safety.
-                          children(Cell, cell, block)
-                        }
-                      </Fragment>
-                    );
-                  })}
-              </Block>
+                {
+                  // @ts-expect-error - expects the consumer of `@ballerine/blocks` to register `Blocks.cells` for type safety.
+                  children(Cell, cell, block)
+                }
+              </Fragment>
             );
-          })}
+          });
+
+          if (!Block) {
+            return <Fragment key={`block:${index}`}>{cellsList}</Fragment>;
```

Comment:
It will generate duplicate keys here. Consider using `useId` or any other mechanism for unique ids.